contracts/MechMarketplace.sol

```diff
@@ -108,26 +108,23 @@ contract MechMarketplace is IErrorsMarketplace {
     mapping(uint8 => address) public mapPaymentTypeBalanceTrackers;
     // Mapping of account nonces
     mapping(address => uint256) public mapNonces;
-    // Set of mechs created by this marketplace
-    address[] public setMechs;
+    // Mapping of service ids to mechs
+    mapping(uint256 => address) public mapServiceIdMech;
 
 
     /// @dev MechMarketplace constructor.
     /// @param _serviceRegistry Service registry contract address.
-    /// @param _stakingFactory Staking factory contract address.
```

Comment:
Dropping all staking stuff. We just need the counters for the activity checkers, not these various variables imo